src/praisonai-agents/praisonaiagents/runtime/registry.py

```
@@ -114,114 +122,266 @@ def clear(self) -> None:
+            if runtime_id in _runtimes:
+                raise ValueError(f"Runtime '{runtime_id}' is already registered")
+            _runtimes[runtime_id] = factory_func
+            _runtime_metadata[runtime_id] = metadata
+    
+    def unregister(self, runtime_id: str) -> bool:
+        """Unregister a runtime."""
+        with _registry_lock:
+            if runtime_id in _runtimes:
+                del _runtimes[runtime_id]
+                if runtime_id in _runtime_metadata:
+                    del _runtime_metadata[runtime_id]
```

Ruff (0.15.17)

[warning] Use `pop` instead of `key in dict` followed by `del dict[key]`

Replace `if` statement with `.pop(..., None)`

(RUF051)